lib/src/utils.dart

```diff
@@ -95,3 +95,19 @@ Logic condFlop(
                 resetValue: resetValue,
                 asyncReset: asyncReset)
             .named('${d.name}_flopped');
+
+/// Swap two [Logic] structures based on a conditional [swap].
+(LogicType, LogicType) logicSwap<LogicType extends Logic>(
+    Logic swap, (LogicType, LogicType) toSwap) {
```

Comment:
Maybe change type to `LogicType` or something since it's not floating-point specific anymore?

Suggestion: why not just call the function `swap`?

Reply:
what's the conflict with `swap`?

yes i was looking at an old version or something, `LogicType` looks good now

Reply:
oh looks like you fixed while i was typing :)